comp/forwarder/eventplatform/eventplatformimpl/epforwarder.go

```diff
@@ -345,20 +346,43 @@ func Diagnose() []diagnose.Diagnosis {
 				Diagnosis: fmt.Sprintf("Connectivity to `%s` is Ok", url),
 			})
 		} else {
-			diagnoses = append(diagnoses, diagnose.Diagnosis{
+			diag := diagnose.Diagnosis{
 				Status:      diagnose.DiagnosisFail,
 				Category:    desc.category,
 				Name:        name,
 				Diagnosis:   fmt.Sprintf("Connection to `%s` failed", url),
 				Remediation: "Please validate Agent configuration and firewall to access " + url,
 				RawError:    err.Error(),
-			})
+			}
+			diag = maybeTestWithPQDN(cfg, endpoints.Main, diag)
```

Comment:
I think it would be more idiomatic to do something like
```go
if cfg.GetBool("convert_dd_site_fqdn.enabled") {
	diag = append(diag, testWithPQDN(cfg, endpoints.Main))
}
```
And remove the check in the function (also rename the function)